libs/x-content/src/test/java/org/elasticsearch/xcontent/XContentParserTests.java

```diff
@@ -69,20 +69,9 @@ public void testFloat() throws IOException {
             assertEquals(value, number.floatValue(), 0.0f);
 
             switch (xContentType) {
-                case VND_CBOR:
-                case VND_SMILE:
-                case CBOR:
-                case SMILE:
-                    assertThat(number, instanceOf(Float.class));
-                    break;
-                case VND_JSON:
-                case VND_YAML:
-                case JSON:
-                case YAML:
-                    assertThat(number, instanceOf(Double.class));
-                    break;
-                default:
-                    throw new AssertionError("unexpected x-content type [" + xContentType + "]");
+                case VND_CBOR, VND_SMILE, CBOR, SMILE -> assertThat(number, instanceOf(Float.class));
+                case VND_JSON, VND_YAML, JSON, YAML -> assertThat(number, instanceOf(Double.class));
+                default -> throw new AssertionError("unexpected x-content type [" + xContentType + "]");
```

Comment:
Should be unnecessary now.

Reply:
Actually, I tried this out, and it's more complicated that I realised. You only seem to get exhaustive checking if the switch returns a value. In this case, it's just calling methods, so we need the default case.